packages/server/lib/controllers/proxy.coffee

```diff
@@ -1,411 +0,0 @@
-        setCookies(true)
-
-        debug("redirecting to new url %o", { status: statusCode, url: newUrl })
-
-        ## finally redirect our user agent back to our domain
-        ## by making this an absolute-path-relative redirect
-        return res.redirect(statusCode, newUrl)
-
-      if headers["x-cypress-file-server-error"]
-        wantsInjection or= "partial"
-
-      setBody(str, statusCode, headers)
-
-    if obj = buffers.take(req.proxiedUrl)
-      wantsInjection = "full"
-
-      onResponse(obj.stream, obj.response)
-    else
-      opts = {
-        timeout: null
-        strictSSL: false
-        followRedirect: false
-        retryIntervals: [0, 100, 200, 200]
-      }
-
-      ## strip unsupported accept-encoding headers
-      encodings = accept.parser(req.headers["accept-encoding"]) ? []
-
-      if "gzip" in encodings
-        ## we only want to support gzip right now
-        req.headers["accept-encoding"] = "gzip"
-      else
-        ## else just delete them since we cannot
-        ## properly decode them
-        delete req.headers["accept-encoding"]
-
-      if remoteState.strategy is "file" and req.proxiedUrl.startsWith(remoteState.origin)
-        opts.url = req.proxiedUrl.replace(remoteState.origin, remoteState.fileServer)
-      else
-        opts.url = req.proxiedUrl
-
-      ## if we have auth headers and this request matches our origin
-      ## protection space and the user has not supplied auth headers
-      if reqNeedsBasicAuthHeaders(req, remoteState)
-        { auth } = remoteState
-
-        base64 = Buffer
-        .from(auth.username + ":" + auth.password)
-        .toString("base64")
-
-        req.headers["authorization"] = "Basic #{base64}"
-
-      rq = request.create(opts)
-
-      rq.on("error", endWithNetworkErr)
-
-      rq.on "response", (incomingRes) ->
-        onResponse(rq, incomingRes)
-
-      ## if our original request has been
-      ## aborted, then ensure we forward
-      ## this onto the proxied request
-      ## https://github.com/cypress-io/cypress/issues/2612
-      ## this can happen on permanent connections
-      ## like SSE, but also on any regular ol'
-      ## http request
-      req.on "aborted", ->
-        rq.abort()
-
-      ## proxy the request body, content-type, headers
-      ## to the new rq
-      req.pipe(rq)
-
-    return thr
-
-  setResHeaders: (req, res, incomingRes, wantsInjection) ->
-    return if res.headersSent
-
-    ## omit problematic headers
-    headers = _.omit incomingRes.headers, "set-cookie", "x-frame-options", "content-length", "content-security-policy", "connection"
-
-    ## do not cache when we inject content into responses
-    ## later on we should switch to an etag system so we dont
-    ## have to download the remote http responses if the etag
-    ## hasnt changed
-    if wantsInjection
-      headers["cache-control"] = "no-cache, no-store, must-revalidate"
-
-    ## proxy the headers
-    res.set(headers)
-}
```

Comment:
Not sure if the history of this file is being tracked in our commits completely. I didn't really investigate. I know it's complicated, but will we be able to find this file in the history of commits from the proxy package?

Reply:
Not really, but the code changes is so drastic and it's split into several files now so I don't think there's even a way to capture this change in Git while keeping file history. Any blame on the proxy package will lead to this PR, and then they can see that the proxy.coffee file was deleted I guess.